src/libs/sfdk/vboxvirtualmachine.cpp

```diff
@@ -429,6 +428,21 @@ void VBoxVirtualMachinePrivate::setVideoMode(const QSize &size, int depth,
         setExtraData(key, value, Sdk::instance(), [=](bool ok) { *allOk &= ok; });
     };
 
+    {
+        BatchComposer composer = BatchComposer::createBatch("setLastNormalWindowPosition");
+        fetchExtraData("GUI/LastNormalWindowPosition", Sdk::instance(), [=, batch = composer.batch()](QString data, bool ok) {
+            if (!ok)
```

Comment:
Let's fix error reporting now. We need

```
*allOk &= ok;
```

prior to this line and then `{ *allOk = false; return; }` instead of just `return` in assertions.

Reply:
That was left out on purpose. Even if this part of setVideoMode fails, the others might be enough, depending on VirtualBox version.

Reply:
Ah, I guess you assume it would cancel the other parths. That's not the case here. It would just indicate that not all went as planned to the user. So I would prefer to be very specific about the two scenarios we expect here, branch for old and new version of vbox we currently know about and be verbose in other cases.

Note: When I tried with vbox 6.1.36r152435, the GUI/LastNormalWindowPosition property is set for my emulators, so this should not fail normally. Also when I tried with nonexisting property, it still returned zero, just the message is "No value set!". Older emulators will get reinstalled this time, because we are touching the emulator package, so our emulators should have that property set after upgrade.

Reply:
> Ah, I guess you assume it would cancel the other parths. That's not the case here. It would just indicate that not all went as planned to the user. So I would prefer to be very specific about the two scenarios we expect here, branch for old and new version of vbox we currently know about and be verbose in other cases.

You are again right. I wonder how I could think that, when the code is right in front of me?

> Note: When I tried with vbox 6.1.36r152435, the GUI/LastNormalWindowPosition property is set for my emulators, so this should not fail normally. Also when I tried with nonexisting property, it still returned zero, just the message is "No value set!". Older emulators will get reinstalled this time, because we are touching the emulator package, so our emulators should have that property set after upgrade.

Thanks for testing!